### ISSUE TITLE
[M1] forward_step_parity passing (fp32, ~1e-4 rel)

### ISSUE THREAD

**Issue by travisgalloway** [travisgalloway/monica#9]
Part of #2.

**Goal:** Prove the training path (`forward`, parallel scan) and inference path (`step`, recurrence) produce the same logits. This catches a silent bug the scan-vs-sequential check does NOT.

**Acceptance criteria**
- `src/conformance/forward_step_parity.py` `check_forward_step_parity` runs green on the toy MLX model in **fp32** (~1e-4 rel).
- Wire the MLX `to_numpy` converter so logits compare correctly.

**Files:** `src/conformance/forward_step_parity.py` (harness exists), `src/model/mlx_backend.py`. Depends on #8.

**Comment by travisgalloway** [comment]
Done — verified on Apple Silicon.

`check_forward_step_parity` (parallel `forward` vs recurrence `step`) passes in fp32 on the toy MLX model (`test_forward_step_parity_toy`).

Verified on Apple Silicon (macOS arm64, Python 3.14.3, mlx 0.31.2) at `main` @ c20f646. Full suite green (20/20) and the M4 smoke gate passes end to end:

```
[reference] step0 loss=8.06889  step49 loss=1.53557
[resumed]   resumed at step=25  step49 loss=1.53557
[match] post-resume max|loss diff| over steps 25..49 = 1.192e-07
[eval] val_loss=1.5416  val_perplexity=4.6721
SMOKE TEST PASSED ✅  resume is exact and eval runs.
```
